fix(ci): Storybook Screenshots race condition on concurrent PR runs

## Observed

When two CI runs execute concurrently on the same PR (e.g. after a fast follow-up commit while the first run is still in progress), both runs attempt to push screenshots to the `gh-screenshots` branch. Because neither run reads the other's push before writing, the second run fails with:

```
##[error]Failed to upload screenshots: Update is not a fast forward
```

The screenshots themselves are captured successfully; only the upload fails. The `yoavf/auto-pr-screenshots-action@v1.4.0-beta` action sets `fail-on-error: true`, so this marks the CI check as failed even though no screenshot content is wrong.

First observed on PR #249 — the initial commit run and the fix commit run both pushed to `pr-249/` simultaneously.

## Expected

The Storybook Screenshots check should not fail due to a concurrent push race. Options include: serializing the upload step, using force-push (accepting that the second run overwrites the first), or upgrading to a version of the action that handles concurrent pushes gracefully.

## File / job

- CI job: `Storybook Screenshots` in `.github/workflows/` (uses `yoavf/auto-pr-screenshots-action@v1.4.0-beta`)
- Branch: `gh-screenshots`

---
*Created by Claude Sonnet 4.6*

--- a/.github/workflows/ci-actions.yml
+++ b/.github/workflows/ci-actions.yml
@@ -54,6 +54,9 @@ jobs:
   storybook-screenshots:
     name: Storybook Screenshots
     if: github.event_name == 'pull_request'
+    concurrency:
+      group: storybook-screenshots-${{ github.event.pull_request.number }}
+      cancel-in-progress: true
     runs-on: ubuntu-latest
     permissions:
       contents: write

